### PR TITLE
handle more error cases with innoextract

### PR DESCRIPTION
**Pull request by Laserlicht**
Changes suggested by @IvanSavenko 
- check for Gog Galaxy installer
- ask for bin file first
- expose innoextract exceptions to user error messagebox

Selected branch is `master`. Hotfix?

### CODE REVIEW

**Comment by Laserlicht** [comment]
@IvanSavenko Any idea why it fails on android NDK?

**Comment by IvanSavenko** [comment]
> @IvanSavenko Any idea why it fails on android NDK?

Looks like `istream_iterator` can't be used with `std;:search`. Perhaps read this data to buffer and compare arrays?

**Comment by Laserlicht** [comment]
> > @IvanSavenko Any idea why it fails on android NDK?
> 
> Looks like `istream_iterator` can't be used with `std;:search`. Perhaps read this data to buffer and compare arrays?

Strange. On linux this wasn't a problem with gcc and clang...
Tried another approach. Hopefully it will compile.

**Comment by Laserlicht** [comment]
@IvanSavenko  Seems to compile now. But there are some other problems. Any idea?

**Comment by IvanSavenko** [comment]
Ah, CI is a bit broken in master due to urgent Google Play upload. Rebase to beta for now.

**Comment by Laserlicht** [comment]
Contains also your both `master` commits... Hopefully it's ok.

**Comment by Laserlicht** [comment]
Seems to work now (tested on android and linux). :)

File: launcher/firstLaunch/firstlaunch_moc.cpp [hunk 1]
```diff
@@ -352,6 +358,33 @@ void FirstLaunchView::extractGogData()
...
+			if(fileSize > 10 * 1024 * 1024)
+				return false; // avoid to load big files; galaxy exe is smaller...
+
+			std::vector<char> buffer;
+			buffer.reserve(fileSize);
+			buffer.insert(buffer.begin(), std::istream_iterator<char>(is), std::istream_iterator<char>());
+
+			std::array<char, 20> magic_id{ 0x47, 0x00, 0x4F, 0x00, 0x47, 0x00, 0x20, 0x00, 0x47, 0x00, 0x61, 0x00, 0x6C, 0x00, 0x61, 0x00, 0x78, 0x00, 0x79, 0x00 }; //GOG Galaxy
+
+			auto res = std::search(buffer.begin(), buffer.end(), magic_id.begin(), magic_id.end());
```
Comment (kambala-decapitator) [inline launcher/firstLaunch/firstlaunch_moc.cpp:381]: is the header located at a random place and not at 0x0?

**Comment by kambala-decapitator** [comment]
QFile + QByteArray would probably be easier to use instead of std classes to search for the header

File: launcher/firstLaunch/firstlaunch_moc.cpp (hunk 1 above)
Comment (Laserlicht) [reply launcher/firstLaunch/firstlaunch_moc.cpp:381]: Yes. At a "random" position. It isn't really a header. It's a String containing "GOG Galaxy". So basically this way it should work with newer versions too.

Comment (kambala-decapitator) [reply launcher/firstLaunch/firstlaunch_moc.cpp:381]: > It's a String containing "GOG Galaxy"

then in code you should use something like `u"GOG Galaxy"` to get utf-16 string and then view it as a byte array

**Pull request merged**